content/index.md

```diff
@@ -0,0 +1,172 @@
+
+> As a developer, I want to receive SMS messages in my application.
+
+> As a developer, I want to buy a number, and have my application respond to incoming calls on this number.
+
+> As a developer, I use ngrok for local development. I don't have a paid account, so I often need to reconfigure my webhook URL (sometimes only one of them).
+
+> As a developer, I am creating an application that sends and receives SMS, using the Messages API for outgoing messages.
+
+> As a developer, I need to inspect the configuration of my account and may also want to change one or more settings.
+
+> As a developer, I haven't used Nexmo since I first looked at it a few weeks ago. I want to look at which applications I have and which numbers are used by which application so that I can decide if I can reallocate or cancel numbers.
+
+> As a developer I want to look up some information about a user's number for security purposes.
+
+> I have been asked to detail for our auditors all the numbers on our Nexmo account and which applications use them. I need to supply this as a spreadsheet.
+
+> As a developer, we run a large number of applications, and each one may have multiple incoming numbers linked to it. I want to know which numbers are linked to application X.
+
+
+## Packaging, Installation and Setup
+
+CLI tools **must**:
+
+ - be available on all platforms (Windows, Mac, Linux - any others would be a bonus)
+ - have as few dependencies as possible
+ - be open to users of all our supported technology stacks (Ruby, PHP, Python, Java, NodeJS, .NET, Go, Android, iOS)
+
+In addition, it would be nice if the tool could:
+
+ - upgrade itself
+ - be extensible with plugins
+ - be relatively small in size
+ - not store user's actual credentials on disk
+
+The tool will be able to state its current version when `--version` is supplied
+
+## Command Design
+
+Consistency and predictability are the key to a command line tools that users become familiar and comfortable with. To this end, all commands will follow the same conventions to enable us to build and maintain a tool that continues to feel familiar.
+
+```
+awesomecli topic:command --kebab-case-param value 
+```
+
+The above shows a basic template to use as a starting point
+
+* **`topic`** will usually be a plural noun. Examples could include `applications`, `numbers`, `insights`.
+* **`command`** will usually be an active or imperative verb. Examples could include `create`, `list`, `configure`.
+* For more complex areas, add a **subtopic** with the format `topic:subtopic:command`. Examples could include `accounts:jwt:generate`, `applications:numbers:link`.
+* All input will be supplied with clearly-named flags each followed by the value for this parameter. The parameters will use "kebab case" and will be named exactly the same when different commands accept the same input data. Examples could include `--number`, `--application`, `--event_url`.
```

Comment:
I'd want to make a distinction between parameters and flag and reserve flags vor variable inputs, like JSON structure properties.

Reply:
Depending on the framework used, it could be as easy as `flags` are before the subcommand and parameters are after. Big 👍 for reserving flag names in general though

Reply:
Specifying the order of flags/subcommands/parameters makes the interface more rigid and more difficult to use, in my opinion. I _have_ used tools which have 'global' flags/options required before the subcommand, and then subcommand specific flags/options _after_ the subcommand, but it's the exception, and I prefer tools where I can tack an option on to the end if I forgot to add it the first time I ran the command.

Reply:
Added a note about reserving global flags (probably the ones mentioned in this doc for a start)